### ISSUE TITLE
1.10 build failure with JDK 11.0.17 when it passes with JDK 11.0.14

### ISSUE THREAD

**Issue by ctubbsii** [apache/accumulo#3068]
**Describe the bug**
I'm not sure what's happening, but after upgrading my environment from JDK 11.0.14 to 11.0.17, a lot of integration tests in the 1.10 branch started failing due to ClassNotFoundException issues (and possibly other problems... it's not clear yet). It does not seem to affect the 2.1 or main branches (but I did verify it also affects 2.0.1). Downgrading back to 11.0.14 seems to resolve the issues.

**Versions (OS, Maven, Java, and others, as appropriate):**
 - Affected version(s) of this project: 1.10.3-SNAPSHOT
 - OS: Fedora 36
 - Others: `java-11-openjdk-devel-1:11.0.14.1.1-5.fc36.x86_64` works, but `java-11-openjdk-devel-1:11.0.17.0.8-2.fc36.x86_64` does not.

**To Reproduce**
Steps to reproduce the behavior (or a link to an example repository that reproduces the problem):
1. Update to the broken version (`dnf upgrade java-11-openjdk\*`)
2. Run `mvn clean verify -Dtest=blah -Dit.test=ClassLoaderIT -Dspotbugs.skip -Dcheckstyle.skip -Dformatter.skip -Dimpsort.skip -Drat.skip`
3. Build fails
4. Downgrade to the working version (`dnf downgrade java-11-openjdk\*`)
5. Run the same `mvn` command
6. Build succeeds

**Expected behavior**
The build should succeed, regardless of the version

**Additional context**
For reproducing, I focused on the ClassLoaderIT test, but I saw the following reliably fail on both the Hadoop2 and Hadoop3 build profiles after updating JDK 11 on the machine:

```
org.apache.accumulo.minicluster.MiniAccumuloClusterTest.testPerTableClasspath
org.apache.accumulo.start.classloader.vfs.AccumuloReloadingVFSClassLoaderTest.testReloading
org.apache.accumulo.start.classloader.vfs.AccumuloVFSClassLoaderTest.testDefaultContextConfigured
org.apache.accumulo.start.classloader.vfs.ContextManagerTest.testPostDelegation
org.apache.accumulo.start.classloader.vfs.ContextManagerTest.differentContexts
org.apache.accumulo.start.classloader.vfs.providers.VfsClassLoaderTest.testGetClass
org.apache.accumulo.test.ShellServerIT.scansWithClassLoaderContext
org.apache.accumulo.test.ShellServerIT.testPertableClasspath
org.apache.accumulo.test.UserCompactionStrategyIT.testPerTableClasspath
org.apache.accumulo.test.functional.ClassLoaderIT.test
org.apache.accumulo.test.functional.ConfigurableCompactionIT.testPerTableClasspath
org.apache.accumulo.test.functional.ScannerContextIT.testOneScannerDoesntInterfereWithAnother
org.apache.accumulo.test.functional.ScannerContextIT.test
org.apache.accumulo.test.functional.ScannerContextIT.testScanContextOverridesTableContext
org.apache.accumulo.test.functional.ScannerContextIT.testClearContext
org.apache.accumulo.test.proxy.TBinaryProxyIT.testCompactionStrategy
org.apache.accumulo.test.proxy.TCompactProxyIT.testCompactionStrategy
org.apache.accumulo.test.proxy.TJsonProtocolProxyIT.testCompactionStrategy
org.apache.accumulo.test.proxy.TTupleProxyIT.testCompactionStrategy
```

It's possible not all these failures are related, but there does seem to be a common theme, in that most of these have something to do with the VfsClassLoader. That was changed slightly in 2.1, so it could explain why the newer branches aren't failing. The other alternative is that the newer branches aren't cross-compiling to Java 8 bytecode like the 1.10 branch is and compile to Java 11 bytecode. However, I would be more inclined to believe it's a classloading issue than a cross-compilation issue.

I'd be curious to know if this was reproducible on Ubuntu, CentOS, or another popular distribution, or with a different distribution of OpenJDK than the one Fedora builds and ships.

**Comment by ctubbsii** [comment]
For what it's worth, I don't see any [relevant bugs reported in Fedora](https://bugzilla.redhat.com/buglist.cgi?bug_status=NEW&bug_status=ASSIGNED&classification=Fedora&product=Fedora&product=Fedora%20EPEL&component=java-11-openjdk). I could create one, but if it can be reproduced outside of Fedora, then reporting it in Fedora is not the right place. It could be a commons-vfs bug also.

**Comment by cshannon** [comment]
@ctubbsii - I can confirm that this fails on RHEL8 as well (but doesn't really tell us much as RHEL8 is based on Fedora).

Anyways, I found the issue, it has to do with the mime type now being properly detected on the jar file as "application/java-archive" and the AccumuloVFSClassLoader.jar class in version 1.10 not having an entry for it.  You actually fixed this already in 2.1 here: https://github.com/apache/accumulo/blob/main/start/src/main/java/org/apache/accumulo/start/classloader/vfs/AccumuloVFSClassLoader.java#L288

**Comment by cshannon** [comment]
And here is the relevant pull request that fixes the issue: https://github.com/apache/accumulo/pull/2683

We just need to backport this to 1.10

**Comment by cshannon** [comment]
@ctubbsii - As you can see I back ported your fix in #3609 by cherry-picking it and tweaking the commit and message slightly. I'm not sure how you usually like to back port fixes (cherry-picking or merging, etc) so feel free to do something else if you don't like the cherry-pick commit.

In terms of what changed, I am not sure. We could try and go through the JDK version history but I don't think it matters too much. Clearly the new JDK 11 version now is returning that new mime type for the file so we should just add it and handle it as adding the new type fixes it either way just like it did for JDK 17.
